include/deal.II/numerics/matrix_tools.h

```diff
@@ -68,6 +69,8 @@ namespace PETScWrappers
 #endif
 
 #ifdef DEAL_II_WITH_TRILINOS
+// TODO: move this to a stand-alone header and include from there?
+#  define TrilinosScalar double
 namespace TrilinosWrappers
```

Comment:
because we forward declare all the trilinos-related objects and I want to define functions with consistent data types.

`TrilinosScalar` seems to be an invention of deal.ii, it is defined in (including this)
```
$ ag --cpp "define TrilinosScalar"
include/deal.II/lac/trilinos_block_sparse_matrix.h
34:#  define TrilinosScalar double

include/deal.II/lac/trilinos_vector_base.h
35:#  define TrilinosScalar double

include/deal.II/lac/trilinos_sparse_matrix.h
36:#  define TrilinosScalar double

include/deal.II/numerics/matrix_tools.h
75:#  define TrilinosScalar double
```
I would say this definition should be moved to a stand-alone header like `lac/trilinos_types.h` and included where needed. What do you think?